### PR TITLE
Require Node.js 10

### PR DESCRIPTION
**Pull request by realityking**
This nicely gets rid of the dependency of `pify`.

### CODE REVIEW

**Comment by realityking** [comment]
@sindresorhus Could you take a look at this one? 🙏

File: index.js [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
+const getOptions = options => ({level: 9, ...options});
 
-module.exports = (input, options) => {
+module.exports = async (input, options) => {
 	if (!input) {
 		return Promise.resolve(0);
 	}
 
-	return pify(zlib.gzip)(input, getOptions(options)).then(data => data.length).catch(_ => 0);
+	const data = await promisify(zlib.gzip)(input, getOptions(options));
```
Comment (sindresorhus) [inline index.js:15]: It's generally better to do the promisification at the top-level if it doesn't require any local state, so it doesn't have to be done for each function invocation.

File: index.js [hunk 2]
```diff
@@ -2,17 +2,18 @@
...
-const pify = require('pify');
 
-const getOptions = options => Object.assign({level: 9}, options);
+const getOptions = options => ({level: 9, ...options});
 
-module.exports = (input, options) => {
+module.exports = async (input, options) => {
 	if (!input) {
 		return Promise.resolve(0);
 	}
```
Comment (sindresorhus) [inline index.js:14]: The `Promise.resolve` is moot now.

File: package.json [hunk 3]
```diff
@@ -30,13 +30,13 @@
...
 	},
 	"devDependencies": {
-		"ava": "^1.4.1",
-		"p-event": "^2.1.0",
-		"tsd": "^0.7.2",
-		"xo": "^0.24.0"
+		"ava": "^2.4.0",
+		"p-event": "^4.2.0",
+		"tsd": "^0.13.1",
+		"typescript": "^4.0.5",
```
Comment (sindresorhus) [inline package.json:39]: This is not needed.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Pull request merged**